example/spec/loadTest/queueGranulePassThroughLoadTest.js

```diff
@@ -0,0 +1,253 @@
+      console.log(redConsoleLog(),
+        `\n${JSON.stringify(granuleError.granuleId)} - ${granuleError.err}`));
+  }
+};
+
+describe('The Granule Ingest Load Test ', () => {
+  beforeAll(async () => {
+    if (!Number.isFinite(Number(granuleCountPerWorkflow)) || granuleCountPerWorkflow > granuleThresholdPerWorkflow) {
+      beforeAllFailed = `===== beforeAll() - Invalid input for number of granules per Workflow detected - ${granuleCountPerWorkflow} =====`;
+      throw new Error(beforeAllFailed);
+    } else {
+      try {
+        config = await loadConfig();
+
+        process.env.GranulesTable = `${config.stackName}-GranulesTable`;
+        process.env.ExecutionsTable = `${config.stackName}-ExecutionsTable`;
+        process.env.CollectionsTable = `${config.stackName}-CollectionsTable`;
+
+        await pMap(
+          new Array(totalWorkflowCount).fill().map((_, index) => index + 1),
+          async (index) => {
+            await batchGranulesProcessing(index);
+          },
+          { concurrency: 8 }
+        );
+      } catch (error) {
+        beforeAllFailed = `===== beforeAll() failed =====\n ${error}`;
+        throw new Error(beforeAllFailed);
+      }
+    }
+  });
+
+  afterAll(async () => {
+    // clean up stack state added by test
+    await Promise.all(
+      totalInputPayloads.map(async (inPayload) =>
+        await cleanupLoadTestGranules(config.stackName, inPayload.granules))
+    );
+
+    console.log(blueConsoleLog(), '\n===== Delete lambdaOutputPayload ExecutionArns =====');
+    queueGranulesChildExecutionArns.forEach((lambdaOutputPayload) => {
+      lambdaOutputPayload.running.forEach((childExecutionArn) => {
+        deleteExecution({
+          prefix: config.stackName,
+          executionArn: childExecutionArn,
+        });
+      });
+    });
+
+    if (queueGranulesExecutionArns !== undefined) {
+      console.log(blueConsoleLog(), '\n===== Delete queueGranules ExecutionArns =====');
+      queueGranulesExecutionArns.forEach((qGranulesExecutionArn) => {
+        deleteExecution({
+          prefix: config.stackName,
+          executionArn: qGranulesExecutionArn,
+        });
+        console.log(blueConsoleLog(), `${JSON.stringify(qGranulesExecutionArn)}`);
+      });
+    }
+
+    await Promise.all(
+      testSuffixes.map(async (testSuffix) => {
+        await cleanupCollections(
+          config.stackName,
+          config.bucket,
+          collectionsDir,
+          testSuffix
+        );
+        cleanupProviders(
+          config.stackName,
+          config.bucket,
+          providersDir,
+          testSuffix
+        );
+      })
+    );
+  });
+
+  it('writes to database the expected number of granules with status completed', () => {
+    const expectedGranuleCount = granuleCountPerWorkflow * totalWorkflowCount;
+    expect(expectedGranuleCount * granuleCountThreshold < totalGranulesCompleted.length && totalGranulesCompleted.length <= expectedGranuleCount).toBeTruthy();
```

Comment:
This test is validating that the granules all completed after some period of time, given some percentage of acceptable?      That seems..... within the parameters of the initial test intent, except we should be representing that *all* of them ingest, or if there's an acceptable failure rate despite retries, I would expect it needs to be much lower than 5% to consider the test a success at this ingest rate.      

There's room to talk about data integrity concerns at load here, but that could probably be acceptably pushed to a downstream epic ticket.     I think the focus of this ticket is looking at tuning/architecture failures, and the data we're providing 

Given we're operating in a multi-datastore environment, I think we need to also evaluate elastic search counts here, and possibly consider validating file records (e.g. granule API return files object) and such.    

For it to be a valid, repeatable test scenario should we document some of this as well?

Reply:
Removed granuleCountThreshold